### PR TITLE
Move deprecated Rounding classes into test

### PR DESCRIPTION
**Pull request by spinscale**
This moves the joda based rounding classes into the test directory, so
it does not get used accidentally. This allows us to keep the duel tests
running for now, which ensure that joda and java time implementations
produce the same results.

### CODE REVIEW

File: benchmarks/src/main/java/org/elasticsearch/benchmark/time/RoundingBenchmark.java [hunk 1]
```diff
@@ -19,7 +19,6 @@
 package org.elasticsearch.benchmark.time;
```
Comment (danielmitterdorfer) [inline benchmarks/src/main/java/org/elasticsearch/benchmark/time/RoundingBenchmark.java]: I wonder whether it makes sense to remove this benchmark entirely now. Is it still beneficial to know about Java time performance without a baseline?

**Review by danielmitterdorfer — COMMENTED** [review]
Looks fine to me; I just left a remark about the necessity of the microbenchmark.

File: benchmarks/src/main/java/org/elasticsearch/benchmark/time/RoundingBenchmark.java (hunk 1 above)
Comment (spinscale) [reply benchmarks/src/main/java/org/elasticsearch/benchmark/time/RoundingBenchmark.java]: as it is in the history we can always revive it. removed.

**Review by danielmitterdorfer — APPROVED** [review]
Thanks! LGTM

**Pull request merged**